M2/Macaulay2/packages/gfanInterface.m2

```diff
@@ -751,14 +751,15 @@ gfanConvertToNewRing (PolynomialRing) := R1 -> (
   --This method does all of the actual conversions.
   --Every other gfanConvertToNewRing uses rings 
   --produced by this method.
-  R1Gens := gens R1;
-  numDigits := length (toString (#R1Gens));
-  R2 := (coefficientRing R1) new Array from (for i in 1..#R1Gens list (
-    value ("x" | demark ("",for i from 1 to numDigits-(length toString i) list "0") | toString i)
-  ) );
-  R2Gens := gens R2;
-  generatorMapping := for i in 0..#(gens R1) - 1 list (R1Gens#i =>R2Gens#i);
-  return (map(R2, R1, generatorMapping), R2);
+  	R1Gens := gens R1;
+  	numDigits := length (toString (#R1Gens));
+	varstr := (for i in 1..#R1Gens list "x" | demark ("",for j from 1 to numDigits-(length toString i) list "0") | toString i);
+	var := new Dictionary;
+	varlist = for i in varstr list getGlobalSymbol(var,i);
+ 	R2 := (coefficientRing R1) new Array from varlist;
+ 	R2Gens := gens R2;
+	generatorMapping := for i in 0..#(gens R1) - 1 list (R1Gens#i =>R2Gens#i);
+	return (map(R2, R1, generatorMapping), R2);
```

Comment:
I think there's something wrong with the indentation here, which makes it look like you edited every line here. Could you revise this commit? (I can show you tomorrow if you don't know how to do this)